src/react/ReactDotplot.js

```diff
@@ -0,0 +1,31 @@
+import React, { useEffect } from "react";
+import DotplotGL from "../index";
+
+const ReactDotplot = ({ id, data, color, size }) => {
+  useEffect(() => {
+    let plot = new DotplotGL(`#${id}`);
+
+    plot.setInput({
+      x: [...data.x],
+      y: [...data.y],
+      xlabels: data.xlabels,
+      ylabels: data.ylabels,
+    });
+
+    plot.setState({
+      size,
+      color,
+    });
+
+    plot.render();
+
+    return () => {
+      plot?.plot.dataWorker.terminate();
+      plot?.plot.webglWorker.terminate();
+    };
+  }, []);
+
+  return <div id={id} style={{ height: "500px", width: "100%" }}></div>;
```

Comment:
just realized this, this probably applies to other react components as well, 

Its preferred to use refs rather than id's so we don't pollute the dom, also ids may conflict with elements. 

an example from kana - https://github.com/jkanche/kana/blob/master/src/components/Plots/uDimPlot.js#L93